yb-voyager/cmd/analyzeSchema.go

```diff
@@ -738,20 +759,38 @@ func createSqlStrInfoArray(path string, objType string) []sqlInfo {
 			}
 		}
 		if dollarQuoteFlag == 2 {
-			if strings.Contains(currLine, ";") {
-				processCollectedSql(path, &stmt, &formattedStmt, objType, &sqlInfoArr, &reportNextSql)
-				//reset for parsing other sqls
-				dollarQuoteFlag = 0
-				codeBlockDelimiter = ""
+			if isEndOfSqlStmt(currLine) {
+				break
 			}
 		}
 	}
-	// check whether there was error reading the script
-	if scanner.Err() != nil {
-		panic(scanner.Err())
+
+	return stmt, formattedStmt
+}
+
+func collectSqlStmt(lines []string, i *int) (string, string) {
+	stmt := ""
+	formattedStmt := ""
+	for ; *i < len(lines); *i++ {
+		currLine := lines[*i]
+		if len(currLine) == 0 {
+			continue
+		}
+
+		stmt += currLine + " "
+		formattedStmt += currLine + "\n"
+
+		if isEndOfSqlStmt(currLine) {
+			break
+		}
 	}
+	return stmt, formattedStmt
+}
 
-	return sqlInfoArr
+func isEndOfSqlStmt(line string) bool {
+	/* not checking for string with suffix `;` to cover cases like comment at the end after `;`
+	   example: "CREATE TABLE t1 (c1 int); -- table t1" */
+	return strings.Contains(line, ";")
```

Comment:
This can potentially be "more" error prone.

I would suggest to trim the trailing comment if exists and then check if the line ends with a semi-colon:
```
x = index of last `--`
if x != -1 {
  line = line[:x]
}
r-trim line | check if ends with semi-colon
```

Reply:
done